docs/modules/ROOT/pages/how-tos/cloudscale/provider-adopt-worker-nodes.adoc

```diff
@@ -0,0 +1,218 @@
+----
+terraform apply
+----
+
+== Apply Machine and MachineSet manifests
+
+[IMPORTANT]
+====
+Please ensure the terraform apply has completed successfully before proceeding with this step.
+Without the tags applied by Terraform, nodes will be duplicated with the same name and weird stuff might happen.
+
+Please be careful to not apply the `MachineSet` before the `Machine` manifests.
+====
+
+. Copy `worker-machines_yml` from the Terraform output and apply it to the cluster.
++
+[source,bash]
+----
+terraform output -raw worker-machines_yml | yq -P > worker-machines.yml
+head worker-machines.yml
+kubectl apply -f worker-machines.yml
+----
+
+. Check that all machines are in the `Running` state.
++
+[source,bash]
+----
+kubectl get -f worker-machines.yml
+----
+
+. Copy `worker-machineset_yml` from the Terraform output and apply it to the cluster.
++
+[source,bash]
+----
+terraform output -raw worker-machineset_yml | yq -P > worker-machineset.yml
+head worker-machineset.yml
+kubectl apply -f worker-machineset.yml
+----
+
+. Copy `infra-machines_yml` from the Terraform output and apply it to the cluster.
++
+[source,bash]
+----
+terraform output -raw infra-machines_yml | yq -P > infra-machines.yml
+head infra-machines.yml
+kubectl apply -f infra-machines.yml
+----
+
+. Check that all machines are in the `Running` state.
++
+[source,bash]
+----
+kubectl get -f infra-machines.yml
+----
+
+. Copy `infra-machineset_yml` from the Terraform output and apply it to the cluster.
++
+[source,bash]
+----
+terraform output -raw infra-machineset_yml | yq -P > infra-machineset.yml
+head infra-machineset.yml
+kubectl apply -f infra-machineset.yml
+----
+
+. Copy `additional-worker-machines_yml` from the Terraform output and apply it to the cluster.
++
+[source,bash]
+----
+terraform output -raw additional-worker-machines_yml | yq -P > additional-worker-machines.yml
+head additional-worker-machines.yml
+kubectl apply -f additional-worker-machines.yml
+----
+
+. Check that all machines are in the `Running` state.
++
+[source,bash]
+----
+kubectl get -f additional-worker-machines.yml
+----
+
+. Copy `additional-worker-machinesets_yml` from the Terraform output and apply it to the cluster.
++
+[source,bash]
+----
+terraform output -raw additional-worker-machinesets_yml | yq -P > additional-worker-machinesets.yml
+head additional-worker-machinesets.yml
+kubectl apply -f additional-worker-machinesets.yml
+----
+
+== Remove nodes from the Terraform state
+
+. Remove the nodes from the Terraform state.
++
+[source,bash]
+----
+terraform state rm module.cluster.module.worker
+terraform state rm module.cluster.module.infra
+terraform state rm module.cluster.module.additional_worker
+cat > override.tf <<EOF
+module "cluster" {
+  infra_count              = 0
+  worker_count             = 0
+  additional_worker_groups = {}
+}
+EOF
+----
+
+. Check the terraform plan output and apply the changes.
+There should be no server recreation.
+Hieradata entries can be ignored.
```

Comment:
```suggestion
Hieradata changes must be ignored, otherwise the cluster ingress controller will become unavailable.
```

We should probably create a follow up to implement support for deactivating ingress backend management via terraform module vshn-lbaas-cloudscale